0.22.1 blogpost (#276)
TeamNewPipe/website: _posts/2022-02-25-newpipe-0.22.1.md

---
layout: post
title: "NewPipe 0.22.1 released: Fixes and adjustments"
short: "NewPipe 0.22.1 released"
date: 2022-02-25 22:00:00 + 01:00
categories: [pinned, release]
author: <a href="https://github.com/Stypox">@Stypox</a>
image: newpipe
excerpt_separator: <!-- more -->
---

Why is there a new release only six days after the previous one? Well, don't expect anything big… just a few adjustments to the previous version and some long-awaited bugfixes for lists! That's what you would expect from a point release, according to [semantic versioning](https://semver.org/), but we have never been that good at choosing release numbers, sorry about that.

<!-- more -->

### Fixed

> - Lists on big screen might finally work as expected! Now if the items loaded initially do not fill the whole screen, NewPipe will load another chunk of them, instead of showing a loading indicator **forever**. This long-awaited fix was brought to you by none other than the hero from last release, [@litetex](https://github.com/litetex). In the same pull request he also fixed some recurring crashes when scrolling through lists, caused by a misuse of the nasty Android `RecyclerView`. 

Reviewer: ```suggestion
- Lists on big screen might finally work as expected! Now if the initially loaded items do not fill the whole screen, NewPipe will load another chunk of them, instead of showing a loading indicator **forever**. This long-awaited fix was brought to you by none other than our hero from last release, [@litetex](https://github.com/litetex). In the same pull request he also fixed some recurring crashes when scrolling through lists, caused by a misuse of the nasty Android `RecyclerView`. 
```

>   > For those of you wondering, a `RecyclerView` is a special *view* component that allows showing many *items* of the same type in a list without overloading the system with too many views at the same time. It does so by binding only the items visible on the screen to the corresponding view component (for example a stream info item view), and when one such view is moved out on one side of the screen it is *recycled* to be used again once needed on the opposite side. The random crash that was fixed here was caused a piece of code that ~~sometimes~~ *once in a blue moon* tried to bind the same view component to multiple items at the same time, which is obviously not possible, hence the crash. Sorry for this out-of-section nerd talk, just wanted to communicate why it took us so long to find out the root cause of the problem.

Reviewer: ```suggestion
  > For those of you wondering, a `RecyclerView` is a special *view* component that allows showing many *items* of the same type in a list without overloading the system with too many views at the same time. It does so by binding only the items visible on the screen to the corresponding view component (for example a stream info item view), and when one such view is moved out on one side of the screen it is *recycled* to be used again once needed on the opposite side. The random crash that was fixed here was caused by a piece of code that *every now and then* tried to bind the same view component to multiple items at the same time, which is obviously not possible, hence the crash. Sorry for this out-of-section nerd talk, we just want to show you how hard it was to track this issue down.
```

> - [@Stypox](https://github.com/Stypox) (I declare freedom from any conflict of interest) solved the graphical problem with the newly-introduced fast-forward/rewind arc not underlaying the system ui when playing in the fullscreen player. This was particularly visible when controls were hidden, since a non-animated border was shown around the animated arc.

Reviewer: ```suggestion
- [@Stypox](https://github.com/Stypox) (I, as the blog author, declare freedom from any conflict of interest) solved the graphical problem with the newly-introduced fast-forward/rewind arc not underlaying the system ui when playing in the fullscreen player. This was particularly visible when controls were hidden, since a non-animated border was shown around the animated arc. This was a really cool fix, very well done [@Stypox](https://github.com/Stypox).
```

Reviewer: Tried to make sure who I was in this context (normally we is used when referring to the blog authors) + I played on the conflict of interest joke a bit more, maybe thats too much though?

Author: I'd prefer not to include it ;-)

image: newpipe
excerpt_separator: <!-- more -->
---

Why is there a new release only six days after the previous one? Well, don't expect anything big… just a few adjustments to the previous version and some long-awaited bugfixes for lists! That's what you would expect from a point release, according to [semantic versioning](https://semver.org/), but we have never been that good at choosing release numbers, sorry about that.

Before starting with the changelog, let us dedicate a brief moment to the suffering of Ukrainian people. We sincerely hope this cruel war will come to an end soon and your freedom will be respected.

<!-- more -->

### Fixed

- Lists on big screen might finally work as expected! Now if the initially loaded items do not fill the whole screen, NewPipe will load another chunk of them, instead of showing a loading indicator **forever**. This long-awaited fix was brought to you by none other than our hero from last release, [@litetex](https://github.com/litetex). In the same pull request they also fixed some recurring crashes when scrolling through lists, caused by a misuse of the nasty Android `RecyclerView`. 

>   > For those of you wondering, a `RecyclerView` is a special *view* component that allows showing many *items* of the same type in a list without overloading the system with too many views at the same time. It does so by binding only the items visible on the screen to the corresponding view component (for example a stream info item view), and when one such view is moved out on one side of the screen it is *recycled* to be used again once needed on the opposite side. The random crash that was fixed here was caused by a piece of code that *every now and then* tried to bind the same view component to multiple items at the same time, which is obviously not possible, hence the crash. Sorry for this out-of-section nerd talk, we just want to show you how hard it was to track this issue down.

Reviewer: ```suggestion
  > For those of you wondering, a `RecyclerView` is a special *view* component that allows showing many *items* of the same type in a list without overloading the system with too many views at the same time. It does so by binding only the items visible on the screen to the corresponding view component (for example a stream info item view), and when one such view is moved out on one side of the screen, it is *recycled* to be used again, once needed on the opposite side. The random crash that was fixed here was caused by a piece of code that *every now and then* tried to bind the same view component to multiple items at the same time, which is obviously not possible, hence the crash. Sorry for this out-of-section nerd talk, we just want to show you how hard it was to track this issue down.
```

- [@Stypox](https://github.com/Stypox) (I, as the blog author, declare freedom from any conflict of interest) solved the graphical problem with the newly-introduced fast-forward/rewind arc not underlaying the system ui when playing in the fullscreen player. This was particularly visible when controls were hidden, since a non-animated border was shown around the animated arc.

> - Talking about system borders… users that have a phone with cutouts (you know, those little notches that phone manufactures really grow to love lately) might have noticed a misplaced player in the previous version. It was caused by an attempted improvement to watching videos in multi-window that evidently went wrong. This change, originally made by [@Cheechaii](https://github.com/Cheechaii), was reverted by [@litetex](https://github.com/litetex). Thanks to both!

Reviewer: ```suggestion
- Talking about system borders… users that have a phone with cutouts (you know, those little notches that phone manufactures really grew to love lately) might have noticed a misplaced player in the previous version. It was caused by an attempt to improve watching videos in multi-window mode that evidently went wrong. This change, originally made by [@Cheechaii](https://github.com/Cheechaii), was reverted by [@litetex](https://github.com/litetex). Thanks to both!
```

this should be the correct time? I think?

## Nerd Talk

- The error catching library, ACRA, was updated by the master of libraries, [@TacoTheDank](https://github.com/TacoTheDank). Thank you!

- [@litetex](https://github.com/litetex) improved the image minimizer GitHub action they introduced a couple of weeks ago to reduce the image size of (phone) screenshots included in issue bodies and comments.

- [@Stypox](https://github.com/Stypox) beat [@TobiGr](https://github.com/TobiGr) on time and got a grip on incrementing Android's `compileSdk` from 30 (Android 11) to 31 (Android 12). This will allow updating more libraries to their newest versions in the future.

> - [@Stypox](https://github.com/Stypox) refactored a couple of lines in the big (we are talking BIG. Really, way **too BIG**) `Player` class. They would also like to point out that the whole team [is](https://github.com/TeamNewPipe/NewPipe/issues/7673) [committed](https://github.com/TeamNewPipe/NewPipe/issues/7939) to improving the messed-up player, which likes to spit out nasty bugs every once in a while due to its complexity. In a very distant future, expressions such as "the player is a hot pile of garbage" and "the code is a spaghetti monster of doom" might finally disappear! This is another step towards that future.

Reviewer: What about adding the self praise as a joke to one of the paragraphs here? Smth like `The blog post author would like to point out here what a great contributor Stypox is.`

Author: Ok, you convinced me, but I think it wouldn't fit too well here, so I will do it on the line discussed before
